src/hooks/useNavigation.ts

```
@@ -0,0 +1,34 @@
+  waitForElements: () => Promise<void>;
+  updateButtonStatus: (url: string) => Promise<void>;
+}
+
+export const useNavigation = ({
+  urlRef,
+  setIsPageReady,
+  waitForElements,
+  updateButtonStatus,
+}: UseNavigationProps) => {
+  useEffect(() => {
+    const handleNavigation = async (event: any) => {
```

Biome

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)